components/org.wso2.carbon.analytics.auth.rest.api/src/main/java/org/wso2/carbon/analytics/auth/rest/api/impl/LoginApiServiceImpl.java

```diff
@@ -194,6 +194,22 @@ public Response loginAppNamePost(String appName
                                 .cookieBuilder(AuthRESTAPIConstants.WSO2_SP_REFRESH_TOKEN, refTokenPart2,
                                         AuthRESTAPIConstants.LOGIN_CONTEXT + appContext, true, true,
                                         refTokenValidityPeriod);
+                        if (IdPClientConstants.REFRESH_GRANT_TYPE.equals(grantType)
```

Comment:
There should be same logic in login.. Move it to a private method and reuse

Reply:
Fixed in 562ea446079da6b14dbc53f63f6d471ca5d917f0